core/templates/vector.h

```diff
@@ -187,6 +187,70 @@ class Vector {
 		return false;
 	}
 
+	struct Iterator {
+		_FORCE_INLINE_ T &operator*() {
+			return *elem_ptr;
+		}
+		_FORCE_INLINE_ T *operator->() { return elem_ptr; }
+		_FORCE_INLINE_ Iterator &operator++() {
+			elem_ptr++;
+			return *this;
+		}
+		_FORCE_INLINE_ Iterator &operator--() {
+			elem_ptr--;
+			return *this;
+		}
+
+		_FORCE_INLINE_ bool operator==(const Iterator &b) const { return elem_ptr == b.elem_ptr; }
+		_FORCE_INLINE_ bool operator!=(const Iterator &b) const { return elem_ptr != b.elem_ptr; }
+
+		Iterator(T *p_ptr) { elem_ptr = p_ptr; }
+		Iterator() {}
+		Iterator(const Iterator &p_it) { elem_ptr = p_it.elem_ptr; }
+
+	private:
+		T *elem_ptr = nullptr;
+	};
+
+	struct ConstIterator {
+		_FORCE_INLINE_ const T &operator*() const {
+			return *elem_ptr;
+		}
+		_FORCE_INLINE_ const T *operator->() const { return elem_ptr; }
+		_FORCE_INLINE_ ConstIterator &operator++() {
+			elem_ptr++;
+			return *this;
+		}
+		_FORCE_INLINE_ ConstIterator &operator--() {
+			elem_ptr++;
```

Comment:
This is another typo I think, should be `elem_ptr--`.

Also, this `ConstIterator` is the only one that returns `ConstIterator &` for increment/decrement operators, other const iterators return `const ConstIterator &`. Since the goal is to be compatible with STL, I think `ConstIterator &` is correct, const and non-const iterator are the same except one uses `T` and the other uses `const T` according to STL.

Reply:
Yeah, I think the increment and decrement operators should return an iterator reference.